src/Core__Object.res

```diff
@@ -1,31 +1,250 @@
 
-@val external is: ('a, 'b) => bool = "Object.is"
+## Examples
+
+```rescript
+let x = Object.empty()
+x->Object.keysToArray->Array.length // 0
+x->Object.get("toString")->Option.isSome // true
+```
+*/
+@obj
+external empty: unit => {..} = ""
+
+/**
+`is` determines if two objects are identical in all contexts. Objects, arrays, records, and other non-primitives are only identical if they reference the **exact** same object in memory. Primitives like ints, floats, and strings are identical if they have the same value. `+0` and `-0` are distinct. NaN is equal to itself. See [Object.is on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is)
+
+In most scenarios use `==` or `===` or the custom `equals` function (if provided) for the type.
+
+## Examples
+
+```rescript
+Object.is(25, 13) // false
+Object.is("abc", "abc") // true
+Object.is(undefined, undefined) // true
+Object.is(undefined, null) // false
+Object.is(-0.0, 0.0) // false
+Object.is(list{1, 2}, list{1, 2}) // false
+
+Object.is([1, 2, 3], [1, 2, 3]) // false
+[1, 2, 3] == [1, 2, 3] // true
+[1, 2, 3] === [1, 2, 3] // false
+
+let fruit = {"name": "Apple" }
+Object.is(fruit, fruit) // true
+Object.is(fruit, {"name": "Apple" }) // false
+fruit == {"name": "Apple" } // true
+fruit === {"name": "Apple" } // false
+```
+*/
+@val
+external is: ('a, 'a) => bool = "Object.is"
 
 @val external create: {..} => {..} = "Object.create"
 @val external createWithProperties: ({..}, {..}) => {..} = "Object.create"
 @val external createWithNull: (@as(json`null`) _, unit) => {..} = "Object.create"
 @val external createWithNullAndProperties: (@as(json`null`) _, {..}) => {..} = "Object.create"
 
-@val external assign: ({..}, {..}) => {..} = "Object.assign"
-@variadic @val external assignMany: ({..}, array<{..}>) => {..} = "Object.assign"
+/**
+`assign(target, source)` copies enumerable own properties from the source to the target, overwriting properties with the same name. It returns the modified target object. A deep clone is not created; properties are copied by reference.
+
+**Note:** ReScript provides [first-class support for immutable objects](https://rescript-lang.org/docs/manual/latest/object), including spreading one object into another. This is often more convenient than using `assign`.  
+
+See [Object.assign on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) or [ECMAScript Language Specification](https://tc39.es/ecma262/multipage/fundamental-objects.html#sec-object.assign).
+
+## Examples
+
+```rescript
+Object.assign({"a": 1}, {"a": 2}) // {"a": 2}
+Object.assign({"a": 1, "b": 2}, {"a": 0}) // {"a": 0, "b": 2}
+Object.assign({"a": 1}, {"a": null}) // {"a": null}
+```
+*/
+@val
+external assign: ({..}, {..}) => {..} = "Object.assign"
```

Comment:
This is actually unsound. It's a very useful function, and there's unfortunately no way to make it sound while maintaining it's usefulness (yet). Therefore I think it ought to be named `assignUnsafe`, and have the unsoundness noted in the docstring.

Reply:
Please see my updated doc warning. I think it is very helpful.

Reply:
This function serves a purpose that the built-in record and object features aren't able to support. That's also why it's not possible to type this function properly. There's really no practical use of this function that is safe. And the only way to encapsulate the unsafety is to explicitly (and correctly) annotate the type of both the arguments and return value on use (or specialization).

This is soundly typed and safe to use.

```rescript
let extend: ({"foo": int}, {"bar": string}) => {"foo": int, "bar": string} = Object.assign
```